app/src/main/java/tech/ula/model/repositories/AppsRepository.kt

```diff
@@ -6,16 +6,29 @@ import android.util.Log
 import tech.ula.model.daos.AppsDao
 import tech.ula.model.entities.App
 import tech.ula.model.remote.RemoteAppsSource
-import tech.ula.utils.asyncAwait
+import tech.ula.utils.AppsPreferences
 
-class AppsRepository(private val appsDao: AppsDao, private val remoteAppsSource: RemoteAppsSource) {
+import tech.ula.utils.asyncAwait
 
+class AppsRepository(
+    private val appsDao: AppsDao,
+    private val remoteAppsSource: RemoteAppsSource,
+    private val appsPreferences: AppsPreferences
+) {
     private val refreshStatus = MutableLiveData<RefreshStatus>()
 
     fun getAllApps(): LiveData<List<App>> {
         return appsDao.getAllApps()
     }
 
+    fun getAppServiceTypePreference(app: App): String {
+        return appsPreferences.getAppServiceTypePreference(app.name)
```

Comment:
Seeing this makes me think this preference should be attached to the app entity rather than used as a Android preference. Should talk about this more on Monday.